### PR TITLE
Riffing on Dennis' idea

### PR DESCRIPTION
**Pull request by martinthomson**

### CODE REVIEW

File: draft-ietf-tls-keylogfile.md [hunk 1]
```diff
@@ -247,6 +247,12 @@ consumption by other programs.  In both cases, applications might require
 special authorization or they might rely on system-level access control to limit
 access to these capabilities.
 
+Forward secrecy guarantees provided in TLS 1.3 (see {{Section 1.2 and Appendix
+E.1 of ?RFC8446}}) and some modes of TLS 1.2 (such as those in {{Sections 2.2
+and 2.4 of ?RFC4492}}) are not provided if keys are saved.  Logged keys
```
Comment (martinthomson) [inline draft-ietf-tls-keylogfile.md]:
```suggestion
and 2.4 of ?RFC4492}}) do not hold if keys are saved.  Logged keys
```

File: draft-ietf-tls-keylogfile.md [hunk 2]
```diff
@@ -247,6 +247,12 @@ consumption by other programs.  In both cases, applications might require
 special authorization or they might rely on system-level access control to limit
 access to these capabilities.
 
+Forward secrecy guarantees provided in TLS 1.3 (see {{Section 1.2 and Appendix
+E.1 of ?RFC8446}}) and some modes of TLS 1.2 (such as those in {{Sections 2.2
+and 2.4 of ?RFC4492}}) do not hold if keys are saved.  Access to logged keys
+affects the authenticity, confidentiality, and integrity of any data that is
+exchanged for longer period than if an authentication key is compromised.
+
```
Comment (dennisjackson) [inline draft-ietf-tls-keylogfile.md]:
```suggestion
Forward secrecy guarantees provided in TLS 1.3 (see {{Section 1.2 and Appendix
E.1 of ?RFC8446}}) and some modes of TLS 1.2 (such as those in {{Sections 2.2
and 2.4 of ?RFC4492}}) do not hold if key material is recorded.  Access to logged material allows an attacker to decrypt data exchanged in previous TLS connections and so undermine the intended security properties of ?RFC8446. 
```

Not sure what the second sentence was going for so word smithed a bit.

**Comment by dennisjackson** [comment]
In the final paragraph of the security considerations, it might be worth clarifying: 

> This draft does not support the export of the resumption_master_secret in TLS1.3 (see {{Section 7.1 of ?RFC8446}}) as exposing this value would impact both the confidentiality and authenticity of data exchanged in future TLS sessions.

File: draft-ietf-tls-keylogfile.md (hunk 2 above)
Comment (martinthomson) [reply draft-ietf-tls-keylogfile.md]:
```suggestion
Forward secrecy guarantees provided in TLS 1.3 (see {{Section 1.2 and Appendix
E.1 of ?RFC8446}}) and some modes of TLS 1.2 (such as those in {{Sections 2.2
and 2.4 of ?RFC4492}}) do not hold if key material is recorded.  Access to key
material allows an attacker to decrypt data exchanged in any logged TLS
connections.
```

?

Comment (dennisjackson) [reply draft-ietf-tls-keylogfile.md]: The aspect that I'm trying to bring out is that we're worried about an attacker getting a key log file in the present and then decrypting previously exchanged data they'd recorded (which is the thing that PFS stops). Your proposed second sentence gets rid of the nuance w.r.t. timing. Perhaps insert 'previously' before logged?

I have visions of these keylog files slowly filling up disks, not unlike nuclear waste, insofar as they remain hazardous indefinitely unless properly disposed of. I don't want to take up another round trip, so happy with whatever formulation you like best.

File: draft-ietf-tls-keylogfile.md [hunk 3]
```diff
@@ -247,6 +247,11 @@ consumption by other programs.  In both cases, applications might require
 special authorization or they might rely on system-level access control to limit
 access to these capabilities.
 
+Forward secrecy guarantees provided in TLS 1.3 (see {{Section 1.2 and Appendix
+E.1 of ?RFC8446}}) and some modes of TLS 1.2 (such as those in {{Sections 2.2
+and 2.4 of ?RFC4492}}) do not hold if key material is recorded.  Access to key
+material allows an attacker to decrypt data exchanged in any logged TLS
+connections.
```
Comment (martinthomson) [inline draft-ietf-tls-keylogfile.md:254]:
```suggestion
connections.

```

File: draft-ietf-tls-keylogfile.md [hunk 4]
```diff
@@ -247,6 +247,12 @@ consumption by other programs.  In both cases, applications might require
 special authorization or they might rely on system-level access control to limit
 access to these capabilities.
 
+Forward secrecy guarantees provided in TLS 1.3 (see {{Section 1.2 and Appendix
+E.1 of ?RFC8446}}) and some modes of TLS 1.2 (such as those in {{Sections 2.2
+and 2.4 of ?RFC4492}}) do not hold if key material is recorded.  Access to key
+material allows an attacker to decrypt data exchanged in any logged TLS
```
Comment (martinthomson) [inline draft-ietf-tls-keylogfile.md]:
```suggestion
material allows an attacker to decrypt data exchanged in any previously logged TLS
```

**Pull request merged**